go/vt/binlog/keyrange_filter.go

```diff
@@ -20,7 +24,7 @@ import (
 func KeyRangeFilterFunc(keyrange *topodatapb.KeyRange, sendReply sendTransactionFunc) sendTransactionFunc {
 	return func(reply *binlogdatapb.BinlogTransaction) error {
 		matched := false
-		filtered := make([]*binlogdatapb.BinlogTransaction_Statement, 0, len(reply.Statements))
+		filtered := []*binlogdatapb.BinlogTransaction_Statement{}
```

Comment:
Why this change? pre-allocating the filtered array with the approximate size sounds better?